dali/kernels/imgproc/pointwise/linear_transformation_cpu.h

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_POINTWISE_LINEAR_TRANSFORMATION_CPU_H_
+#define DALI_KERNELS_IMGPROC_POINTWISE_LINEAR_TRANSFORMATION_CPU_H_
+
+#include <vector>
+#include <utility>
+#include "dali/core/format.h"
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/common/block_setup.h"
+#include "dali/kernels/imgproc/surface.h"
+#include "dali/kernels/imgproc/roi.h"
+
+namespace dali {
+namespace kernels {
+
+template <typename OutputType, typename InputType,
+        int channels_out, int channels_in, int spatial_ndims>
+class LinearTransformationCpu {
+ private:
+  static constexpr auto ndims_ = spatial_ndims + 1;
```

Comment:
how about going the other way around (from ndims to spatial_dims) as it is what was done in other kernels?